### ISSUE TITLE
invalid character 'c' looking for beginning of value

### ISSUE THREAD

**Issue by danielmapar** [jthomperoo/custom-pod-autoscaler#136]
**Describe the bug**
Currently using the CPA in a project, but it keeps returning `E0423 02:11:09.508568       1 main.go:273] invalid character 'c' looking for beginning of value` after deploying my `evaluate` and `metric` functions.

**To Reproduce**
My code is available [here](https://github.com/danielmapar/EECS6446_Project/tree/main/custom-autoscaler/cpa).

I reproduced this issue with both version `1.0.3` and `1.1.0` of the operator.

It needs extra setup to work (mongo, prometheus, etc..). On that note all I can say is that `metric.py` is working properly and sending `json` data to `evaluate.py`. `evaluate.py` is also working properly.

if I remove `sys.stdout.write(json.dumps(evaluation))` from `evaluate.py` and do:

```python
sys.stderr.write(json.dumps(evaluation))
exit(1)
```
My output is: 
```sh
I0423 02:19:32.448712       1 main.go:231] Waiting 1 milliseconds before starting autoscaler
I0423 02:19:32.450211       1 main.go:290] Starting API using HTTP
I0423 02:19:32.450585       1 main.go:240] Starting autoscaler
I0423 02:19:44.226080       1 shell.go:84] Shell command failed, stderr: {"targetReplicas": 2}
E0423 02:19:44.226185       1 main.go:273] exit status 1
```

However, if I keep `sys.stdout.write(json.dumps(evaluation))` my output is:

```sh
I0423 02:10:57.562897       1 main.go:231] Waiting 1 milliseconds before starting autoscaler
I0423 02:10:57.563356       1 main.go:290] Starting API using HTTP
I0423 02:10:57.564856       1 main.go:240] Starting autoscaler
E0423 02:11:09.508568       1 main.go:273] invalid character 'c' looking for beginning of value
E0423 02:11:19.314626       1 main.go:273] invalid character 'c' looking for beginning of value
E0423 02:11:29.334700       1 main.go:273] invalid character 'c' looking for beginning of value
E0423 02:11:39.343757       1 main.go:273] invalid character 'c' looking for beginning of value
```

**Expected behavior**
The following error should not be showing.

**Kubernetes Details (`kubectl version`):**
```sh
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2+k3s1", GitCommit:"1d4adb0301b9a63ceec8cabb11b309e061f43d5f", GitTreeState:"clean", BuildDate:"2021-01-14T23:52:37Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by danielmapar** [comment]
After cracking my head a bit I discovered the issue. I was calling `print` in one of my inner functions and messing out with `stout`.

**Comment by jthomperoo** [comment]
Hey, glad you got it fixed! The error it gives is not great, it would be better if the error was more descriptive and actually printed out the stdout that was causing it - I've raised this as a new issue here: #137
